doc/hooks/versionCheckHook.section.md

```diff
@@ -16,20 +22,65 @@ stdenv.mkDerivation (finalAttrs: {
+| Rebuild after modification                | Modifying `versionCheckHook` triggers the package rebuild.                                                                                                              | `tests.version` does not rebuild the package, since it runs after package build time.                                                        |
+| Overhead during package build time        | Negligible. Although executed during the the package build time, `versionCheckHook` is lean and executes few commands.                                                  | Zero, since `tests.version` is a derivation that runs after package build time.                                                              |
+| Failure detection time                    | `versionCheckHook` runs during the package build time, therefore it does not deal with failures happening after that.                                                   | `tests.version` detects failures after package build time, like rewritings and relocations.                                                  |
+| Package breakage awareness                | Loud and clear as soon as `versionCheckHook` is reached during the package build time.                                                                                  | Requires specific commands to be noticed, e.g. `nix-build -A pkg.tests.version`.                                                             |
+| OfBorg CI                                 | `versionCheckHook` will be executed by OfBorg, since ofBorg automatically executes the packages affected by the pull request (PR).                                      | OfBorg supports automatic execution of `passthru.tests` for _packages directly affected_ by the PR.                                          |
+| OfBorg CI, transitive dependencies        | `versionCheckHook` will executed by OfBorg, since it builds transitive dependencies of a package affected by the PR. (????)                                             | OfBorg does not support automatic execution of `passthru.tests` for transitive dependencies,  requiring extra commands for this.             |
+| `nixpkgs-review`                          | `versionCheckHook` will be executed by `nixpkgs-review`.                                                                                                                | `nixpkgs-review` has no support for executing `passthru.tests` at all. [^1]                                                                  |
+| `nixpkgs-review`, transitive dependencies | `versionCheckHook` will be executed by `nixpkgs-review`, since the tool reaches transitive dependencies automatically.                                                  | `nixpkgs-review` has no support for executing `passthru.tests` at all,  even for transitive dependencies. [^1]                               |
```

Comment:
Here it is weird to split these to 2 lines, as nixpkgs-review always acts upon all affected packages.